tests/units/test_openstackmap.py

```diff
@@ -0,0 +1,291 @@
+                   ' "links": {"self": ' \
+                   '"http://cloud.host.fi-ware.org:4731/v3/projects/00000000000000000000000000000001"}, "enabled":' \
+                   ' true, "id": "00000000000000000000000000000001", "domain_id": "default", "name": "user cloud"}]}'
+            resp._content = text
+            resp.status_code = 200
+
+        elif url == '/role_assignments':
+            text = '{"role_assignments": [{"scope": {"project": {"id": "00000000000000000000000000000001"}},' \
+                   ' "role": {"id": "00000000000000000000000000000001"}, "user": {"id":' \
+                   ' "00000000000000000000000000000001"}, "links": {"assignment": ' \
+                   '"http://cloud.host.fi-ware.org:4731/v3/projects/00000000000000000000000000000001' \
+                   '/users/00000000000000000000000000000001"}}]}'
+            resp._content = text
+            resp.status_code = 200
+
+        return resp
+
+
+class TestOpenstackMap(TestCase):
+
+    mock_session = MySessionMock()
+
+    def setUp(self):
+        self.OS_AUTH_URL = 'http://cloud.host.fi-ware.org:4731/v2.0'
+        self.OS_USERNAME = 'user'
+        self.OS_PASSWORD = 'password'
+        self.OS_TENANT_NAME = 'user cloud'
+        self.OS_TENANT_ID = '00000000000000000000000000000001'
+        self.OS_REGION_NAME = 'Spain2'
+        self.OS_TRUST_ID = ''
+        self.OS_KEYSTONE_ADMIN_ENDPOINT = 'http://cloud.lab.fiware.org:4730'
+
+        environ.setdefault('OS_AUTH_URL', self.OS_AUTH_URL)
+        environ.setdefault('OS_USERNAME', self.OS_USERNAME)
+        environ.setdefault('OS_PASSWORD', self.OS_PASSWORD)
+        environ.setdefault('OS_TENANT_NAME', self.OS_TENANT_NAME)
+        environ.setdefault('OS_TENANT_ID', self.OS_TENANT_ID)
+        environ.setdefault('OS_REGION_NAME', self.OS_REGION_NAME)
+        environ.setdefault('OS_TRUST_ID', self.OS_TRUST_ID)
+
+    @patch.object(os, 'mkdir')
+    @patch.object(os.path, 'exists')
+    def test_implement_openstackmap(self, mock_exists, mock_os):
+        """test_implement_openstackmap check that we could build an empty map from the resources (VMs, networks, images,
+        volumes, users, tenants, roles...) in an OpenStack infrastructure."""
+        mock_exists.return_value = False
+        mock_os.return_value = True
+        openstackmap = OpenStackMap(region=self.OS_REGION_NAME, auto_load=False)
+        self.assertTrue(mock_os.called)
+        self.assertIsNotNone(openstackmap)
+
+    def test_implement_openstackmap_without_region_name(self):
+        """test_implement_openstackmap_without_region_name check that we could not build an empty map without providing
+        a region Name."""
+
+        del os.environ['OS_REGION_NAME']
+        try:
+            myopenstackmap = OpenStackMap(auth_url=self.OS_AUTH_URL, auto_load=False)
+        except Exception as ex:
+            self.assertRaises(ex)
```

Comment:
To check that an exception is raised when callable is called, I think following way is better than using _try/except_ blocks as part of unit tests:

       with self.assertRaises(Exception):
            OpenStackMap(auth_url=self.OS_AUTH_URL, auto_load=False)